src/org/thoughtcrime/securesms/WelcomeActivity.java

```diff
@@ -60,6 +61,16 @@ public void onCreate(Bundle bundle) {
         DcEventCenter eventCenter = DcHelper.getEventCenter(this);
         eventCenter.addObserver(DcContext.DC_EVENT_CONFIGURE_PROGRESS, this);
         eventCenter.addObserver(DcContext.DC_EVENT_IMEX_PROGRESS, this);
+
+        if (!DcHelper.hasAnyConfiguredContext(this)) {
+          Intent intent = new Intent();
+          // Since android API 26 only explicit broadcasts are allowed for IPC with a few exceptions.
```

Comment:
Btw. a workaround for the need to define the package names of all companion apps we support could be to use a share intent. It has some interesting implications, eg. if the user *would* have installed several companion apps, the user could manually select which one to choose for the account setup. 
Broadcast receivers listening for share intents can also be registered in the AndroidManifest. When registered there, the app can be started from the system - in contrast to broadcast receivers that are registered on runtime. 
However I didn't went this path because the share intent is not inteded to be used like that (from my point of view) and it felt more hackish.

Reply:
Sounds a lot nicer to me than hardcoding the list of companion apps. And letting the user manually select which companion app to use for the account setup does sound like the way to go.

With "share intent", do you mean "an intent with ACTION_SEND"? Or "an intent with ACTION_VIEW"?

/me also wondering why it's called "Broadcast" if you can only send it to one explicitly-given app, but anyway...

Reply:
i commented on that "share alternative" at https://github.com/deltachat/deltachat-android/pull/2088#discussion_r746072280 - in short, i would stay with broadcasts as well for now.